src/main/java/nl/delftelectronics/spaceinvaders/core/GameInformation.java

```diff
@@ -0,0 +1,311 @@
+        }
+    }
+
+    /**
+     * Save the game information to the filename.
+     *
+     * @param os OutputStream to write to
+     */
+    public void save(OutputStream os) {
+        try {
+            ObjectOutputStream oos = new ObjectOutputStream(os);
+            oos.writeObject(this);
+        } catch (IOException e) {
+            Logger.error("The save game file could not be written to. Do you have the correct"
+                    + "permissions? Or perhaps you're trying to serialize an unserializable "
+                    + "object?");
+        }
+    }
+
+    /**
+     * Load the game information from the default save game file location,
+     *
+     * @return the loaded game information
+     */
+    public static Optional<GameInformation> load() {
+        return load(SAVEGAME_FILENAME);
+    }
+
+    /**
+     * Load the game information from the filename.
+     *
+     * @param filename filename of the save game
+     * @return the loaded game information
+     */
+    public static Optional<GameInformation> load(String filename) {
+        try {
+            File file = new File("." + File.separator, filename);
+            FileInputStream fin = new FileInputStream(file);
+            return load(fin);
+        } catch (FileNotFoundException e) {
+            Logger.error("The save game file could not be found.");
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Load the game information from the filename.
+     *
+     * @param is input stream of the game information
+     * @return the loaded game information
+     */
+    public static Optional<GameInformation> load(InputStream is) {
+        Optional<GameInformation> gi = Optional.empty();
+        try {
+            ObjectInputStream ois = new ObjectInputStream(is);
+            gi = Optional.of((GameInformation) ois.readObject());
+        } catch (IOException e) {
+            Logger.error("The save game file could not be accessed.");
+        } catch (ClassNotFoundException e) {
+            Logger.error("Apparently, a class that is stored in the save file is not available. "
+                    + "Perhaps the save file is from an older version of the game.");
+        }
+        return gi;
+    }
+
+    /**
+     * Increment the current level.
+     */
+    public void incrementLevel() {
+        level++;
+    }
+
+    /**
+     * Returns true iff lives is not zero.
+     *
+     * @return true iff lives is not zero
+     */
+    public boolean hasLives() {
+        return lives != 0;
+    }
+
+    /**
+     * Increment the number of lives.
+     */
+    public void incrementLives() {
+        lives++;
+    }
+
+    /**
+     * Decrement the number of lives.
+     */
+    public void decrementLives() {
+        lives--;
+    }
+
+    /**
+     * Returns true iff the player has any bombs.
+     *
+     * @return true iff the player has any bombs.
+     */
+    public boolean hasBombs() {
+        return bombs != 0;
```

Comment:
Do you mean `bombs >=0`? I trust the ship here, because if the ship tries to decrement the bombs without even checking if the player has any bombs (whether or not we include a 0 check), the ship would try to add a bomb to the scene any way.